source/paddle.js

```diff
@@ -425,55 +449,64 @@ paddle.Node = class {
                 let value = '?';
                 let visible = true;
                 let type = null;
-                switch (attr.type) {
-                    case paddle.AttributeType.STRING:
-                        type = 'string';
-                        value = attr.s;
-                        break;
-                    case paddle.AttributeType.STRINGS:
-                        type = 'string[]';
-                        value = Array.from(attr.strings);
-                        break;
-                    case paddle.AttributeType.BOOLEAN:
-                        type = 'boolean';
-                        value = attr.b;
-                        break;
-                    case paddle.AttributeType.BOOLEANS:
-                        type = 'boolean[]';
-                        value = attr.bools ? Array.from(attr.bools) : attr.bools;
-                        break;
-                    case paddle.AttributeType.FLOAT:
-                        type = 'float32';
-                        value = attr.f;
-                        break;
-                    case paddle.AttributeType.FLOATS:
-                        type = 'float32[]';
-                        value = attr.floats ? Array.from(attr.floats) : attr.floats;
-                        break;
-                    case paddle.AttributeType.FLOAT64:
-                        type = 'float64';
-                        value = attr.float64;
-                        break;
-                    case paddle.AttributeType.FLOAT64S:
-                        type = 'float64[]';
-                        value = attr.float64s ? Array.from(attr.float64s) : attr.float64s;
-                        break;
-                    case paddle.AttributeType.INT:
-                        type = 'int32';
-                        value = attr.i;
-                        break;
-                    case paddle.AttributeType.INTS:
-                        type = 'int32[]';
-                        value = attr.ints ? Array.from(attr.ints) : attr.ints;
-                        break;
-                    case paddle.AttributeType.LONG:
-                        type = 'int64';
-                        break;
-                    case paddle.AttributeType.LONGS:
-                        type = 'int64[]';
-                        break;
-                    default:
-                        break;
+
+                if (attr instanceof paddle.IR.Attr) {
+                    type = attr.type;
+                    value = attr.value;
+                } else if (attr instanceof paddle.IR.Region) {
```

Comment:
Similarly, could this be solved differently to avoid `paddle.IR.Region` dependencies and special casing.

Reply:
Done!